### PR TITLE
[MO] - restrict KRaft for Kafka 2.8.1

### PR DESCRIPTION
**Pull request by see-quick**
This PR restrict usage of KRaft for Kafka `2.8.1` and it's allowed only for `3.x.x` Kafka versions. Moreover, I have added tests cases for sanity checks. 

Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### CODE REVIEW

File: src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java [hunk 1]
```diff
@@ -92,8 +93,15 @@ private StrimziKafkaContainer(CompletableFuture<String> imageName) {
...
-            imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(kafkaVersion));
+        if (!this.imageNameProvider.isDone()) {
+            this.imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(this.kafkaVersion));
+        }
+        try {
+            if (this.useKraft && ((this.kafkaVersion != null && this.kafkaVersion.equals("2.8.1")) || this.imageNameProvider.get().contains("2.8.1"))) {
+                throw new UnsupportedKraftKafkaVersionException("Specified Kafka version " + this.kafkaVersion + " is not supported in KRaft mode.");
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error("Error occurred during retrieving of image name provider", e);
```
Comment (scholzj) [inline src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java:104]: I guess this should throw some other exception?

File: src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java [hunk 2]
```diff
@@ -92,8 +93,15 @@ private StrimziKafkaContainer(CompletableFuture<String> imageName) {
 
     @Override
     protected void doStart() {
-        if (!imageNameProvider.isDone()) {
-            imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(kafkaVersion));
+        if (!this.imageNameProvider.isDone()) {
+            this.imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(this.kafkaVersion));
+        }
```
Comment (scholzj) [inline src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java:98]: Why is this first? I would expect the check which does nothing and just throws exception to be first before calling some other methods. Or is it this method which sets the fields used below?

File: src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java (hunk 1 above)
Comment (see-quick) [reply src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java:104]: This exception is possibly thrown when I invoke `this.imageNameProvider.get().` That's why it is needed here

File: src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java (hunk 2 above)
Comment (see-quick) [reply src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java:98]: >  Or is it this method which sets the fields used below?

Yes :)

File: src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java (hunk 1 above)
Comment (scholzj) [reply src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java:104]: Sure, but why you just log it and ignore it? It clearly suggests the check didn't complete and there might be some issue.

Comment (see-quick) [reply src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java:104]: Ah, I see your point. I can add line, which will printStackTrace (`e.printStackTrace();`)

**Review by scholzj — APPROVED** [review]

**Pull request merged**